docs/internal/types.md

```diff
@@ -67,7 +67,57 @@ Most importantly, capabilities are not type traits like C++'s `const int * const
 
 ## Module / Class model
 
-TODO
+In Verona, modules are classes.
+Classes and functions inside modules are just sub-types and methods of the module's class.
+
+However, unlike classes, modules cannot be created or destroyed, so they cannot be allocated in some region.
+Therefore, modules cannot have fields (which must be initialised upon construction), and so Verona has no global variables.
```

Comment:
If modules can be instantiated, then they should have the same semantics as classes. However, the root module (where Main is) is technically instantiated at startup and now we need a way to specify instantiation of its fields, and allow mutable fields that act like global variables.

Of course, due to regions, only a single behaviour can update those "global variables" at one given time, but now methods can have side-effects if they change global context without making it explicit.

I think the fact that I originally had a very weak understanding of this concept and that now I think I understand even less, is a testament that we need more discussions in this area.

Reply:
They're just classes.  If the `Main` function is like a C++ static (doesn't have a parameter that is a pointer to the module) then it won't need (or be allowed to take) a pointer to an instance of the module as a parameter, so you won't need to instantiate the module to invoke `Main`.  The `Main` method may instantiate a module, but it may also instantiate any other class.

Reply:
ok, I have removed that entire paragraph.